js/coinegg.js

```diff
@@ -258,36 +255,14 @@ module.exports = class coinegg extends Exchange {
 
     async fetchTickers (symbols = undefined, params = {}) {
         await this.loadMarkets ();
-        let quoteIds = this.options['quoteIds'];
+        const markets = Object.values (this.markets_by_id);
         let result = {};
-        for (let b = 0; b < quoteIds.length; b++) {
-            let quoteId = quoteIds[b];
-            let tickers = await this.webGetQuoteAllcoin ({
-                'quote': quoteId,
-            });
-            let baseIds = Object.keys (tickers);
-            if (!baseIds.length) {
-                throw new ExchangeError ('fetchTickers failed');
-            }
-            for (let i = 0; i < baseIds.length; i++) {
-                let baseId = baseIds[i];
-                let ticker = tickers[baseId];
-                let id = baseId + quoteId;
-                if (id in this.markets_by_id) {
-                    let market = this.marketsById[id];
-                    let symbol = market['symbol'];
-                    result[symbol] = this.parseTicker ({
-                        'high': ticker[4],
-                        'low': ticker[5],
-                        'buy': ticker[2],
-                        'sell': ticker[3],
-                        'last': ticker[1],
-                        'change': ticker[8],
-                        'vol': ticker[6],
-                        'quoteVol': ticker[7],
-                    }, market);
-                }
+        for (let i = 0; i < markets.length; i++) {
+            const market = markets[i];
+            if (symbols !== undefined && symbols.indexOf (market['symbol']) < 0) {
+                continue;
             }
+            result[market['symbol']] = await this.fetchTicker (market['symbol'], params);
```

Comment:
Oops )

I suggest to drop `fetchTickers`, because by a rule of thumb we stick to one https request per unified call (so methods can be combined in userland, and always behave in a more-or-less unified fashion). We very rarely send more than one call per method, only if we absolutely have to and it cannot be done otherwise. 

Also, we usually don't invent `fetchTickers`, if the exchange in question does not have it. So, in this case the [exchange metainfo](https://github.com/ccxt/ccxt/wiki/Manual#exchange-metadata) for would contain `has['fetchTickers'] == 'emulated'`. 

However, due to rate-limiting and other factors, we don't emulate `fetchTickers` with multiple consequent calls looping over all markets with `fetchTicker`. It would be hard to account for all possible use-cases, retry-strategies and so on...  

Instead, we just tell the user the actual truth, which is: `exchange.has['fetchTickers'] == false && exchange.has['fetchTicker'] == true` and let the user wrap `fetchTicker` how he wants )

Reply:
Makes sense, updating with `fetchTickers` removed as suggested.

Reply:
Updated, hope I got it right

Reply:
Thx for your help! Appreciate it very much!